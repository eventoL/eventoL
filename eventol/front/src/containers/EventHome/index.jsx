import React from 'react';
import PropTypes from 'prop-types';
import withSizes from 'react-sizes';

import Hero from '../../components/Hero';
import Header from '../../components/Header';
import Search from '../../components/Search';
import TitleList from '../../components/TitleList';
import {
  getSearchUrl, getMyEventsUrl,
  getUpcommingEventsUrl,
  getFinishedEventsUrl,
} from '../../utils/urls';
import {
  MOBILE_WIDTH,
  BACKGROUND_DEFAULT,
  LOGO_HEADER_DEFAULT,
  LOGO_LANDING_DEFAULT,
} from '../../utils/constants';

import './index.scss';


class EventHome extends React.Component {
  static propTypes = {
    isMobile: PropTypes.bool.isRequired,
    background: PropTypes.string,
    logoHeader: PropTypes.string,
    logoLanding: PropTypes.string,
    tagMessage: PropTypes.string,
    tagSlug: PropTypes.string,
    user: PropTypes.shape({
      first_name: PropTypes.string,
      last_name: PropTypes.string,
    }),
  }

  static defaultProps = {
    background: BACKGROUND_DEFAULT,
    logoHeader: LOGO_HEADER_DEFAULT,
    logoLanding: LOGO_LANDING_DEFAULT,
    tagMessage: null,
    tagSlug: null,
    user: null,
  }

  state = {
    searchTerm: '',
    searchUrl: '',
    searched: false,
  }

  handleOnEnter = () => {
    const {tagSlug} = this.props;
    const {searchTerm} = this.state;
    if (searchTerm !== ''){
      const searchUrl = getSearchUrl(searchTerm, tagSlug);
      this.setState({searchUrl, searched: true});
    }
  }

  handleOnChange = searchTerm => this.setState({searchTerm})

  render(){
    const {searched, searchUrl} = this.state;
    const {
      user, tagSlug, background,
      logoHeader, logoLanding,
      tagMessage, isMobile,
    } = this.props;
    return (
      <div>
        <Header logoHeader={logoHeader} user={user} isMobile={isMobile} />
        <Hero background={background} logoLanding={logoLanding} message={tagMessage} slug={tagSlug}>
          <Search onChange={this.handleOnChange} onEnter={this.handleOnEnter} />
        </Hero>
        {searched && <TitleList showEmpty id='search_results' title={gettext('Search results')} url={searchUrl} />}
        <TitleList
          id='my_events'
          title={gettext('My Events')}
          url={getMyEventsUrl(tagSlug)}
        />
        <TitleList
          id='next'
          title={gettext('Upcoming Events')}
          url={getUpcommingEventsUrl(tagSlug)}
        />
        <TitleList
          id='finished'
          title={gettext('Finished Events')}
          url={getFinishedEventsUrl(tagSlug)}
        />
      </div>
    );
  }
}

const mapSizesToProps = ({width}) => ({
  isMobile: width < MOBILE_WIDTH,
});

export default withSizes(mapSizesToProps)(EventHome);
