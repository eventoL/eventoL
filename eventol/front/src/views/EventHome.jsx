import React from 'react';
import {render} from 'react-dom';
import {Provider} from 'react-redux';

import getStore from '../store';
import EventHome from '../containers/EventHome';

import './index.scss';

const store = getStore();

window.render_components = properties => {
  window.params = {...properties};
  render(
    (
      <Provider store={store}>
        <EventHome
          background={properties.background}
          logoHeader={properties.logo_header}
          logoLanding={properties.logo_landing}
          tagMessage={properties.message}
          tagSlug={properties.tag_slug}
          user={properties.user}
        />
      </Provider>
    ), document.getElementById('root'),
  );
};

if (module.hot){
  if (window.params) window.render_components(window.params);
  module.hot.accept();
}
