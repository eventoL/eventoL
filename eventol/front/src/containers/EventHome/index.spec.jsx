jest.mock('react-sizes', () => () => Component => props => <Component {...props} />);

jest.mock('../../components/Hero', () => 'Hero');
jest.mock('../../components/Header', () => 'Header');
jest.mock('../../components/Search', () => 'Search');
jest.mock('../../components/TitleList', () => 'TitleList');

import React from 'react';
import renderer from 'react-test-renderer';

import EventHome from '.';


describe('EventHome', () => {
  let component, tree, background,
    logoHeader, logoLanding, tagSlug,
    tagMessage, user, props;

  const getComponent = allProps => {
    component = renderer.create(<EventHome {...allProps} />);
    return component;
  };

  beforeEach(() => {
    tagSlug = 'tagSlug';
    background = 'background';
    logoHeader = 'logoHeader';
    logoLanding = 'logoLanding';
    tagMessage = 'eventolMessage';
    user = {
      first_name: 'first_name',
      last_name: 'last_name',
    };
    props = {
      user,
      tagSlug,
      background,
      logoHeader,
      logoLanding,
      tagMessage,
    };
  });

  describe('With user', () => {
    describe('Desktop', () => {
      describe('Default', () => {
        beforeEach(() => {
          component = getComponent({user, isMobile: false, tagSlug});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });

      describe('With custom data', () => {
        beforeEach(() => {
          component = getComponent({...props, isMobile: false});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });
    });

    describe('Mobile', () => {
      describe('Default', () => {
        beforeEach(() => {
          component = getComponent({user, isMobile: true, tagSlug});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });

      describe('With custom data', () => {
        beforeEach(() => {
          component = getComponent({...props, isMobile: true});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });
    });
  });

  describe('Without user', () => {
    describe('Desktop', () => {
      describe('Default', () => {
        beforeEach(() => {
          component = getComponent({isMobile: false, tagSlug});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });

      describe('With custom data', () => {
        beforeEach(() => {
          props = {
            tagSlug,
            tagMessage,
            background,
            logoHeader,
            logoLanding,
          };
          component = getComponent({...props, isMobile: false});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });
    });

    describe('Mobile', () => {
      describe('Default', () => {
        beforeEach(() => {
          component = getComponent({isMobile: true, tagSlug});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });

      describe('With custom data', () => {
        beforeEach(() => {
          props = {
            tagSlug,
            tagMessage,
            background,
            logoHeader,
            logoLanding,
          };
          component = getComponent({...props, isMobile: true});
          tree = component.toJSON();
        });

        test('Snapshot', () => {
          expect(tree).toMatchSnapshot();
        });
      });
    });
  });
});
