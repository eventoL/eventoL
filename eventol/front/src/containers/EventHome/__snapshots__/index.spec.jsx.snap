// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventHome With user Desktop Default Snapshot 1`] = `
<div>
  <Header
    isMobile={false}
    logoHeader="/static/manager/img/eventol-white.png"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="/static/manager/img/background.png"
    logoLanding="/static/manager/img/logo.png"
    message={null}
    slug="tagSlug"
  >
    <Search
      onChange={[Function]}
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
</div>
`;

exports[`EventHome With user Desktop With custom data Snapshot 1`] = `
<div>
  <Header
    isMobile={false}
    logoHeader="logoHeader"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="background"
    logoLanding="logoLanding"
    message="eventolMessage"
    slug="tagSlug"
  >
    <Search
      onChange={[Function]}
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
</div>
`;

exports[`EventHome With user Mobile Default Snapshot 1`] = `
<div>
  <Header
    isMobile={true}
    logoHeader="/static/manager/img/eventol-white.png"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="/static/manager/img/background.png"
    logoLanding="/static/manager/img/logo.png"
    message={null}
    slug="tagSlug"
  >
    <Search
      onChange={[Function]}
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
</div>
`;

exports[`EventHome With user Mobile With custom data Snapshot 1`] = `
<div>
  <Header
    isMobile={true}
    logoHeader="logoHeader"
    user={
      Object {
        "first_name": "first_name",
        "last_name": "last_name",
      }
    }
  />
  <Hero
    background="background"
    logoLanding="logoLanding"
    message="eventolMessage"
    slug="tagSlug"
  >
    <Search
      onChange={[Function]}
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
</div>
`;

exports[`EventHome Without user Desktop Default Snapshot 1`] = `
<div>
  <Header
    isMobile={false}
    logoHeader="/static/manager/img/eventol-white.png"
    user={null}
  />
  <Hero
    background="/static/manager/img/background.png"
    logoLanding="/static/manager/img/logo.png"
    message={null}
    slug="tagSlug"
  >
    <Search
      onChange={[Function]}
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
</div>
`;

exports[`EventHome Without user Desktop With custom data Snapshot 1`] = `
<div>
  <Header
    isMobile={false}
    logoHeader="logoHeader"
    user={null}
  />
  <Hero
    background="background"
    logoLanding="logoLanding"
    message="eventolMessage"
    slug="tagSlug"
  >
    <Search
      onChange={[Function]}
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
</div>
`;

exports[`EventHome Without user Mobile Default Snapshot 1`] = `
<div>
  <Header
    isMobile={true}
    logoHeader="/static/manager/img/eventol-white.png"
    user={null}
  />
  <Hero
    background="/static/manager/img/background.png"
    logoLanding="/static/manager/img/logo.png"
    message={null}
    slug="tagSlug"
  >
    <Search
      onChange={[Function]}
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
</div>
`;

exports[`EventHome Without user Mobile With custom data Snapshot 1`] = `
<div>
  <Header
    isMobile={true}
    logoHeader="logoHeader"
    user={null}
  />
  <Hero
    background="background"
    logoLanding="logoLanding"
    message="eventolMessage"
    slug="tagSlug"
  >
    <Search
      onChange={[Function]}
      onEnter={[Function]}
    />
  </Hero>
  <TitleList
    id="my_events"
    title="My Events"
    url="?my_events=true&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="next"
    title="Upcoming Events"
    url="?registration_is_open=true&ordering=last_date&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
  <TitleList
    id="finished"
    title="Finished Events"
    url="?registration_is_open=false&ordering=-attendees_count&fields=event_slug,place,image,name,attendees_count,abstract,tags&tags__slug=tagSlug"
  />
</div>
`;
